functionary/ui/views/teams.py

```diff
@@ -0,0 +1,20 @@
+from django.views.generic.detail import DetailView
+from django.views.generic.list import ListView
+
+from core.models import Environment, Team, TeamUserRole
+
+
+class TeamListView(ListView):
+    model = Team
+
+
+class TeamDetailView(DetailView):
+    model = Team
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["environments"] = Environment.objects.filter(team=self.get_object())
```

Comment:
I made a similar comment elsewhere, but you can get at the Environment and TeamUserRole objects through the Team:

```python
team = self.get_object()
context["environments"] = team.environments.all()
context["users"] = [ user_role.user for user_role in team.user_roles.all() ]
```

Reply:
Updated.